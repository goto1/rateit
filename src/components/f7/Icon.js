import React from "react";
import PropTypes from "prop-types";

const Icon = ({ material, className }) => {
  const classNames = className
    ? `material-icons icon ${className}`
    : `material-icons icon`;
  return (
    <i className={classNames}>
      {material}
    </i>
  );
};

Icon.propTypes = {
  material: PropTypes.string.isRequired,
  classNames: PropTypes.string
};

export default Icon;
