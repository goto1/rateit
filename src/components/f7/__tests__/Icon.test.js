import React from "react";
import Icon from "../Icon";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-enzyme";

describe("Icon component", () => {
  const iconName = "thumbs_down";

  it("renders correctly", () => {
    const tree = renderer.create(<Icon material={iconName} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with custom `className` property", () => {
    const className = "customStyles";
    const wrapper = shallow(<Icon material={iconName} className={className} />);
    expect(wrapper).toHaveClassName(className);
  });

  it("renders name of the icon", () => {
    const wrapper = shallow(<Icon material={iconName} />);
    expect(wrapper).toHaveText(iconName);
  });
});
