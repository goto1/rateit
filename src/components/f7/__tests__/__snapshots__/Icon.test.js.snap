// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component renders correctly 1`] = `
<i
  className="material-icons icon"
>
  thumbs_down
</i>
`;
